Tests/StreamChatTests/Database/DatabaseContainer_Tests.swift

```diff
@@ -193,6 +119,45 @@ final class DatabaseContainer_Tests: XCTestCase {
             }
         }
     }
+    
+    func test_removingAllData_whileAnotherWrite() throws {
```

Comment:
Can we maybe add a test or try in our Demo App to logout while there is another controller reference alive and then access that controller's data? 🤔 It would be interesting to see if it previously crashed, and now it does not. Although it might be tricky to actually reproduce this 🤔

Reply:
I think I should be able to create a test for this.

1. Create controller
2. Write some data
3. Call remove all
4. When remove is done, accessing controller content

Reply:
Added a test what fails without current changes.